bundles/org.openhab.binding.homewizard/src/main/java/org/openhab/binding/homewizard/internal/handler/HomeWizardP1MeterHandler.java

```diff
@@ -87,40 +88,34 @@ protected void handleDataPayload(DataPayload payload) {
-            updateState(HomeWizardBindingConstants.CHANNEL_TOTAL_GAS,
+        updateState(HomeWizardBindingConstants.CHANNEL_ANY_POWER_FAILURES,
+                new DecimalType(payload.getAnyPowerFailCount()));
+        updateState(HomeWizardBindingConstants.CHANNEL_LONG_POWER_FAILURES,
+                new DecimalType(payload.getLongPowerFailCount()));
+
+        updateState(HomeWizardBindingConstants.CHANNEL_ACTIVE_CURRENT,
+                new QuantityType<>(payload.getActiveCurrent(), Units.AMPERE));
+        updateState(HomeWizardBindingConstants.CHANNEL_ACTIVE_CURRENT_L1,
+                new QuantityType<>(payload.getActiveCurrentL1(), Units.AMPERE));
+        updateState(HomeWizardBindingConstants.CHANNEL_ACTIVE_CURRENT_L2,
+                new QuantityType<>(payload.getActiveCurrentL2(), Units.AMPERE));
+        updateState(HomeWizardBindingConstants.CHANNEL_ACTIVE_CURRENT_L3,
+                new QuantityType<>(payload.getActiveCurrentL3(), Units.AMPERE));
+
+        updateState(HomeWizardBindingConstants.CHANNEL_ACTIVE_VOLTAGE,
+                new QuantityType<>(payload.getActiveVoltage(), Units.VOLT));
+        updateState(HomeWizardBindingConstants.CHANNEL_ACTIVE_VOLTAGE_L1,
+                new QuantityType<>(payload.getActiveVoltageL1(), Units.VOLT));
+        updateState(HomeWizardBindingConstants.CHANNEL_ACTIVE_VOLTAGE_L2,
+                new QuantityType<>(payload.getActiveVoltageL2(), Units.VOLT));
+        updateState(HomeWizardBindingConstants.CHANNEL_ACTIVE_VOLTAGE_L3,
+                new QuantityType<>(payload.getActiveVoltageL3(), Units.VOLT));
+
+        ZonedDateTime gasTimestamp = payload.getGasTimestamp();
+        if (gasTimestamp != null) {
+            updateState(HomeWizardBindingConstants.CHANNEL_GAS_TOTAL,
                     new QuantityType<>(payload.getTotalGasM3(), SIUnits.CUBIC_METRE));
-
-            // 210119164000
-            int seconds = (int) (dtv % 100);
-
-            dtv /= 100;
-            int minutes = (int) (dtv % 100);
-
-            dtv /= 100;
-            int hours = (int) (dtv % 100);
-
-            dtv /= 100;
-            int day = (int) (dtv % 100);
-
-            dtv /= 100;
-            int month = (int) (dtv % 100);
-
-            dtv /= 100;
-            int year = (int) (dtv + 2000);
-
-            try {
-                DateTimeType dtt = new DateTimeType(
-                        ZonedDateTime.of(year, month, day, hours, minutes, seconds, 0, ZoneId.systemDefault()));
-                updateState(HomeWizardBindingConstants.CHANNEL_GAS_TIMESTAMP, dtt);
-                updateState(HomeWizardBindingConstants.CHANNEL_TOTAL_GAS,
-                        new QuantityType<>(payload.getTotalGasM3(), SIUnits.CUBIC_METRE));
-            } catch (DateTimeException e) {
-                logger.warn("Unable to parse Gas timestamp: {}", payload.getGasTimestamp());
-            }
+            updateState(HomeWizardBindingConstants.CHANNEL_GAS_TIMESTAMP, new DateTimeType(gasTimestamp));
```

Comment:
Same as before, but consider using `TimeZoneProvider` for getting openHAB time-zone rather than system default.

Reply:
Done